sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -694,4 +708,63 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
 
     return !!result?.id;
   }
+
+  /**
+   * Update the program active/inactive state.
+   * @param programId program to be updated.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   * @param isActive value to update the isActive state for the program.
+   * @param options method options.
+   * - `institutionId` institution used for authorization.
+   * - `notes` notes associated with the change.
+   * @returns update result.
+   */
+  async updateEducationProgramIsActive(
+    programId: number,
+    auditUserId: number,
+    isActive: boolean,
+    options?: {
+      institutionId?: number;
+      notes?: string;
+    },
+  ): Promise<UpdateResult> {
+    // Ensures the program exists and the institution has access to it.
+    const program = await this.getInstitutionProgram(
+      programId,
+      options?.institutionId,
+    );
+    if (!program) {
+      throw new CustomNamedError(
+        "Not able to find the education program.",
+        EDUCATION_PROGRAM_NOT_FOUND,
+      );
+    }
+    // Avoid overriding the record with new auditing values if the isActive is already the expected one.
+    if (program.isActive === isActive) {
+      throw new CustomNamedError(
+        "The education program is already set as requested.",
+        EDUCATION_PROGRAM_INVALID_OPERATION,
+      );
+    }
+    return this.dataSource.transaction(async (entityManager) => {
+      if (options?.notes) {
+        await this.institutionService.createInstitutionNote(
+          program.institution.id,
+          NoteType.Program,
+          options.notes,
+          auditUserId,
+          entityManager,
+        );
+      }
+      const now = new Date();
+      const auditUser = { id: auditUserId } as User;
+      return entityManager.getRepository(EducationProgram).update(program.id, {
+        isActive,
+        isActiveUpdatedBy: auditUser,
+        isActiveUpdatedOn: now,
+        updatedAt: now,
```

Comment:
Do we need to explicitly update this? I see that there is a default value `now()` set for this column in db.

Reply:
The default values will be set from DB, as you said. Because this is also the same value as the `isActiveUpdatedOn` the idea of using `now` is to enforce that the DB will receive the exact timestamp.